hawkbit-rest/hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtTargetTypeResource.java

```diff
@@ -0,0 +1,149 @@
+import org.eclipse.hawkbit.repository.DistributionSetTypeManagement;
+import org.eclipse.hawkbit.repository.EntityFactory;
+import org.eclipse.hawkbit.repository.OffsetBasedPageRequest;
+import org.eclipse.hawkbit.repository.TargetTypeManagement;
+import org.eclipse.hawkbit.repository.exception.EntityNotFoundException;
+import org.eclipse.hawkbit.repository.model.TargetType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Slice;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * REST Resource handling for {@link TargetType} CRUD operations.
+ */
+@RestController
+public class MgmtTargetTypeResource implements MgmtTargetTypeRestApi {
+    private static final Logger LOG = LoggerFactory.getLogger(MgmtTargetTypeResource.class);
+
+    private final TargetTypeManagement targetTypeManagement;
+    private final EntityFactory entityFactory;
+
+    public MgmtTargetTypeResource(TargetTypeManagement targetTypeManagement, final EntityFactory entityFactory) {
+        this.targetTypeManagement = targetTypeManagement;
+        this.entityFactory = entityFactory;
+    }
+
+    @Override
+    public ResponseEntity<PagedList<MgmtTargetType>> getTargetTypes(
+            @RequestParam(value = MgmtRestConstants.REQUEST_PARAMETER_PAGING_OFFSET, defaultValue = MgmtRestConstants.REQUEST_PARAMETER_PAGING_DEFAULT_OFFSET) final int pagingOffsetParam,
+            @RequestParam(value = MgmtRestConstants.REQUEST_PARAMETER_PAGING_LIMIT, defaultValue = MgmtRestConstants.REQUEST_PARAMETER_PAGING_DEFAULT_LIMIT) final int pagingLimitParam,
+            @RequestParam(value = MgmtRestConstants.REQUEST_PARAMETER_SORTING, required = false) final String sortParam,
+            @RequestParam(value = MgmtRestConstants.REQUEST_PARAMETER_SEARCH, required = false) final String rsqlParam) {
+
+        final int sanitizedOffsetParam = PagingUtility.sanitizeOffsetParam(pagingOffsetParam);
+        final int sanitizedLimitParam = PagingUtility.sanitizePageLimitParam(pagingLimitParam);
+        final Sort sorting = PagingUtility.sanitizeTargetTypeSortParam(sortParam);
+        final Pageable pageable = new OffsetBasedPageRequest(sanitizedOffsetParam, sanitizedLimitParam, sorting);
+
+        final Slice<TargetType> findTargetTypesAll;
+        long countTargetTypesAll;
+        if (rsqlParam != null) {
+            findTargetTypesAll = targetTypeManagement.findByRsql(pageable, rsqlParam);
+            countTargetTypesAll = ((Page<TargetType>) findTargetTypesAll).getTotalElements();
```

Comment:
Unnecessary cast due to bound assignment to `findTargetTypesAll`. Introduce a variable to store result of `targetTypeManagement.findByRsql(pageable, rsqlParam)` instead, this can be called directly for total elements and can also be assigned to the slice.
E.g.
```java
Page<TargetType> findTargetTypePage = targetTypeManagement.findByRsql(pageable, rsqlParam)
```